### PR TITLE
Add custom events extension tracking in feature statsbeat

### PR DESCRIPTION
**Pull request by lzchen**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py [hunk 1]
```diff
@@ -215,6 +219,10 @@ def _get_feature_metric(self, options: CallbackOptions) -> Iterable[Observation]
         if not self._meets_long_interval_threshold(_FEATURE_METRIC_NAME[0]):
             return observations
         # Feature metric
+        # Check if any features were enabled during runtime
+        if get_statsbeat_custom_events_feature_set():
+            self._feature |= _StatsbeatFeature.CUSTOM_EVENTS_EXTENSION
+            _StatsbeatMetrics._FEATURE_ATTRIBUTES["feature"] = self._feature
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py:225]: Are we currently only checking for updates on custom events?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py [hunk 2]
```diff
@@ -393,6 +401,10 @@ def _get_exception_count(self, options: CallbackOptions) -> Iterable[Observation
                     _REQUESTS_MAP[_REQ_EXCEPTION_NAME[1]][code] = 0 # type: ignore
         return observations
 
+    def set_custom_events_extension_feature(self):
+        self._feature |= _StatsbeatFeature.CUSTOM_EVENTS_EXTENSION
+        _StatsbeatMetrics._FEATURE_ATTRIBUTES["feature"] = self._feature
+
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py:407]: This seems unused. Older design?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py (hunk 1 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py:225]: I will add these in a different pr if needed.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py (hunk 2 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_statsbeat_metrics.py:407]: Removed.

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**